### PR TITLE
Imxrt/timer: Support to set clock source and calculate clock frequency in status structure

### PR DESCRIPTION
**Pull request by jeongarmy**

### CODE REVIEW

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 1]
```diff
@@ -249,6 +249,10 @@ static int imxrt_gpt_getstatus(struct timer_lowerhalf_s *lower,
 {
 	struct imxrt_gpt_lowerhalf_s *priv = (struct imxrt_gpt_lowerhalf_s *)lower;
 	uint32_t ticks;
+#if defined(CONFIG_ARCH_BOARD_IMXRT1020_EVK) || defined(CONFIG_ARCH_BOARD_IMXRT1050_EVK)
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: Let's remove this unnecessary line.

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 2]
```diff
@@ -271,7 +275,12 @@ static int imxrt_gpt_getstatus(struct timer_lowerhalf_s *lower,
 	ticks = imxrt_gpt_getcurrenttimercount(priv->gpt->base);
 	status->timeleft = imxrt_gpt_convert_time_tick(TICK2TIME, priv->config.clockSource, ticks);
 #if defined(CONFIG_ARCH_BOARD_IMXRT1020_EVK) || defined(CONFIG_ARCH_BOARD_IMXRT1050_EVK)
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: same as above

**Comment by seinfra** [comment]
Target : [3909aa3dd857769913812b0b6744580236a41c40] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [3909aa3dd857769913812b0b6744580236a41c40] - Code Rule Check (C++) OK.

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 3]
```diff
@@ -271,7 +275,12 @@ static int imxrt_gpt_getstatus(struct timer_lowerhalf_s *lower,
...
-	status->clock_freq = imxrt_gpt_get_clockfreq(priv->config.clockSource);
+	if (clock_source == kGPT_ClockSource_Osc) {
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: clock_source -> priv->config.clockSource

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 4]
```diff
@@ -416,6 +416,14 @@ static int imxrt_gpt_ioctl(struct timer_lowerhalf_s *lower, int cmd,
 		ret = up_prioritize_irq(priv->gpt->irq_id, arg);
 		break;
 #endif
+	case TCIOC_SETCLKSRC:
+		if (arg < kGPT_ClockSource_Off || arg > kGPT_ClockSource_Osc) {
+			ret = -EINVAL;
+		} else {
+			imxrt_gpt_setclocksource(priv->gpt->base, (gpt_clock_source_t)arg);
+			ret = OK;
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: Please add below
```
priv->config.clockSource = (gpt_clock_source_t)arg;
```

**Comment by seinfra** [comment]
Target : [d2fb2c568a0bcd47ccb31e01d4aba17e3d6c5980] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [d2fb2c568a0bcd47ccb31e01d4aba17e3d6c5980] - Code Rule Check OK.

File: os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c [hunk 5]
```diff
@@ -416,6 +421,15 @@ static int imxrt_gpt_ioctl(struct timer_lowerhalf_s *lower, int cmd,
 		ret = up_prioritize_irq(priv->gpt->irq_id, arg);
 		break;
 #endif
+	case TCIOC_SETCLKSRC:
+		if (arg < kGPT_ClockSource_Off || arg > kGPT_ClockSource_Osc) {
```
Comment (sunghan-chang) [inline os/arch/arm/src/imxrt/imxrt_timer_lowerhalf.c]: line 379
```
int ret = -EINVAL;
```
here
```
case TCIOC_SETCLKSRC:
    if (arg > kGPT_ClockSource_Off && arg <= kGPT_ClockSource_Osc) {
        imxrt_gpt_setclocksource(xxx);
        ....
        ret = OK;
    }
```

**Comment by seinfra** [comment]
Target : [2bc5cd5dc0a0c9275fc47e7da62db30890c4d4ef] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [2bc5cd5dc0a0c9275fc47e7da62db30890c4d4ef] - Code Rule Check OK.

**Pull request merged**